stores/upload.ts

```diff
@@ -0,0 +1,72 @@
+import { defineStore } from 'pinia'
+
+export interface UploadFileData {
+  epubMetadata?: {
+    epubFileName: string
+    title: string
+    author: string
+    language: string
+    coverData: string
+    description: string
+    tags: string[]
+    thumbnailArweaveId: string
+    thumbnailIpfsHash: string
+  }
+  fileRecords: Array<{
+    fileName: string
+    fileType: string
+    arweaveId: string
+    arweaveLink: string
+    ipfsHash: string
+    arweaveKey?: string
+  }>
+  iscnRecord?: any
+}
+
+export const useUploadStore = defineStore('upload', {
+  state: () => ({
+    uploadFileData: null as UploadFileData | null
+  }),
+
+  actions: {
+    setUploadFileData (data: Partial<UploadFileData>) {
+      this.uploadFileData = {
+        ...this.uploadFileData,
+        ...data
+      }
```

Comment:
Might want to split `set` and `update` for more robustness
Maybe add a flag to control behaviour

Reply:
[🎨 Refactor upload file data handling](https://github.com/likecoin/nft-book-press/pull/292/commits/3b5fa862cd66f2e612c6ba88c6ec208b6f1bd7da)